Another batch of SA2.0 edits in TS2.0 (part 3)

jdavcs opened the pull request:
SQLAlchemy 2.0 compatibility. Follow up to #16799. Ref #12541.

## How to test the changes?
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

jdavcs committed:
Fix SA2.0 usage, refactor tool_shed.webapp.api.repository_revisions

jdavcs committed:
Fix SA2.0 usage in tool_shed.webapp.api.groups

jdavcs committed:
Fix SA2.0 usage in tool_shed.webapp.controllers.user

Also, add case_sensitive parameter to shared function in galaxy user manager

jdavcs committed:
Fix SA2.0 usage in tool_shed.webapp.controllers.repository

jdavcs committed:
Fix SA2.0 usage in tool_shed.webapp.controllers.repository

jdavcs committed:
Fix SA2.0 formatting in tool_shed.repository_registry + refactor

jdavcs committed:
Fix SA2.0 usage in tool_shed.test.base.twilltestcase

jdavcs committed:
Drop unused functions from tool_shed.test.base.test_db_util

jdavcs committed:
Fix SA2.0 usage in tool_shed.test.base.test_db_util; refactor

jdavcs committed:
Fix SA2.0 usage in tool_shed.managers.categories

jdavcs committed:
Fix SA2.0 usage in tool_shed.managers.users

jdavcs committed:
Fix SA2.0 usage in tool_shed.managers.repositories

jdavcs committed:
Fix SA2.0 usage in tool_shed.managers.groups

jdavcs committed:
Fix SA2.0 usage in tool_shed.util.metadata_util

jdavcs committed:
Fix SA2.0 usage in tool_shed.util.shed_util_common

jdavcs committed:
Fix SA2.0 usage in tool_shed.util.shed_index

jdavcs committed:
Fix SA2.0 usage in tool_shed.util.search_util

jdavcs committed:
Fix SA2.0 usage in tool_shed.util.commit_util

jdavcs committed:
Fix SA2.0 usage in tool_shed.util.repository_util

jdavcs committed:
Fix SA2.0 usage in tool_shed.metadata, refactor

bgruening commented on lib/tool_shed/metadata/repository_metadata_manager.py:
```diff
@@ -1107,3 +1066,39 @@ def _get_changeset_revisions_that_contain_tools(app: "ToolShedApp", repo, reposi
     return changeset_revisions_that_contain_tools
+
+
+def get_user_by_username(session, username, user_model):
```
Why is the user_model here the third and below it always the second argument?
Does the model really need to be passed in? Can we not assume it has the user model by the function name?

jdavcs replied:
Hmm.. I passed all those models to these functions because in a different module directly importing them from `tool_shed.webapp.model` raised an error due to a circular dependency. However, those were modules in `util`, but this is in `managers`, so there should be no error from an import. Let me clean this up a bit (I hate passing those models).

jdavcs replied:
I am not reusing the function from `galaxy.managers.users` because that one calls `first()`, whereas this one calls `one()`, so the behavior is slightly different. Most likely the difference is inconsequential, but with the tool shed I cannot be sure. And the calling code would have to be slightly adjusted. Given the scale of these changes, I'm trying to minimize my footprint here.

jdavcs committed:
Import models

jdavcs committed:
Ignore typing error (see note)

RepositoryMetadata is mapped imperatively, so until the mapper creates
attributes based on Column definitions, the attribute does not exist on
the class. This class is mapped imperatively due to the complications
triggered by mapping the "metadata" attribute.
